### PR TITLE
Allow specifying opts file & test directory 

### LINKED ISSUE #95: Add more options to mocha-casper.opts
I noticed the mocha-casperjs.opts file behaved differently from the mocha.opts file :

- Mocha looks for the opts file in the `/test` directory
- Mocha allows specifying a opts file path: `mocha --opts test/unit/mocha.opts`

Whereas mocha-casperjs.opts must be in the current directory or it won't be found (unless I'm mistaken).

Also, mocha.opts allows specifying a test directory where to look for test files:
mocha.opts:
`test/unit`

Any opinion ?

### PR DESCRIPTION
**Pull request by windkomo**
Usage :

Option file path :

```
mocha-casperjs --opts=path/to/file/file.opts
```

Add test directory in the opts file:

```
test/casperjs/tests
[other options]
```

Fix #95

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -156,16 +157,20 @@ if (opts.slow) {
 }
 
 // load the user's tests
-var tests = []
+var tests = [],
+    extraArgs = cli.parse((fs.read(optsFile).split('\n'))).args;
```
Comment (nathanboktae) [inline bin/cli.js]: You're assuming the `optsFile` exists here, and that it has a test directory specified, thus breaking all the tests.

**Comment by nathanboktae** [comment]
Sorry for not responding earlier but I paused since all the tests failed.

Also add a test for this feature with something like

```coffee-script
      it 'should use an opts file from another location', (done) ->
        fs.writeFileSync 'mocha-casperjs.opts', '--user-agent=wrong-optsfile'
        fs.writeFileSync 'some-other-mocha-casperjs.opts', '--user-agent=some-other-mocha-casperjs-opts'
        runMochaCasperJsTest
          params: ['--reporter=json']
          test: ->
            casper.page.settings.userAgent.should.equal 'some-other-mocha-casperjs-opts'
        , (output, code) ->
          results = JSON.parse output
          results.stats.passes.should.equal 1
          results.stats.failures.should.equal 0
          done()
```

**Comment by windkomo** [comment]
@nathanboktae Thanks for the feedback, I made some changes that I hope will suit you

File: bin/cli.js [hunk 2]
```diff
@@ -157,15 +160,18 @@ if (opts.slow) {
...
+
 if (cliOptions.args.length > 1) {
   // use tests if they specified them explicty
   tests = cliOptions.args.slice()
   tests.shift()
 } else {
-  // otherwise, load files from the test or tests directory like Mocha does
-  var testDir = 'test'
+  // otherwise, load files from the opts directory, test or tests directory like Mocha does
+  var testDir = extraArgs.length && extraArgs[0] || null
```
Comment (nathanboktae) [inline bin/cli.js]: `var testDir = extraArgs[0]` is sufficient (undefined and null are both falsy)

**Comment by nathanboktae** [comment]
Thanks for the good work :)

**Pull request merged**